source/traction_boundary_conditions/lithostatic_pressure.cc

```diff
@@ -0,0 +1,469 @@
+           p = bvs.begin();
+           p != bvs.end(); ++p)
+        {
+          if (p->second.get() == this)
+            traction_bi.insert(p->first);
+        }
+      AssertThrow(*(traction_bi.begin()) != numbers::invalid_boundary_id,
+                  ExcMessage("Did not find the boundary indicator for the lithostatic pressure plugin."));
+
+      // the below is adapted from adiabatic_conditions/initial_profile.cc
+      // but we use the initial temperature and composition and only calculate
+      // the pressure profile
+
+      // the spacing of the depth profile
+      delta_z = this->get_geometry_model().maximal_depth() / (n_points-1);
+
+      const unsigned int n_compositional_fields = this->n_compositional_fields();
+
+      // the pressure at the surface
+      pressure[0]    = this->get_surface_pressure();
+
+      // For spherical(-like) domains, modify the representative point
+      Point<dim> spherical_representative_point(representative_point);
+      const double degrees_to_radians = dealii::numbers::PI/180.0;
+
+      // check location of representative point and
+      // set radius to radius of surface of domain
+      // for spherical domains
+      if (const GeometryModel::SphericalShell<dim> *gm = dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()))
+        {
+          spherical_representative_point[1] *= degrees_to_radians;
+
+          AssertThrow(spherical_representative_point[1] >= 0.0 &&
+                      spherical_representative_point[1] <= gm->opening_angle(),
+                      ExcMessage("Longitude of representative point outside shell domain."));
+
+          // set outer radius
+          spherical_representative_point[0] = gm->outer_radius();
+
+          if (dim == 3)
+            spherical_representative_point[2] *= degrees_to_radians;
+        }
+      else if (const GeometryModel::Chunk<dim> *gm = dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()))
+        {
+          spherical_representative_point[1] *= degrees_to_radians;
+
+          AssertThrow(spherical_representative_point[1] >= gm->west_longitude() &&
+                      spherical_representative_point[1] <= gm->east_longitude(),
+                      ExcMessage("Longitude of representative point outside chunk domain."));
+
+          if (dim ==3)
+            {
+              spherical_representative_point[2] *= degrees_to_radians;
+
+              AssertThrow(spherical_representative_point[2] >= gm->south_latitude() &&
+                          spherical_representative_point[2] <= gm->north_latitude(),
+                          ExcMessage("Latitude of representative point outside chunk domain."));
+            }
+
+          // set outer radius
+          spherical_representative_point[0] = gm->outer_radius();
+        }
+//      TODO There will be a LayeredChunk soon :)
```

Comment:
The layered chunk logic will be the same as for the chunk, right? So you can just put them into one if-statement and save some code.